app/src/test/adminUsers.test.tsx

```diff
@@ -1,12 +1,421 @@
+    });
+
+  //wait for page to be done loading  
+  await waitFor(() => {
+    expect(screen.queryByTestId('ball-triangle-loading')).not.toBeInTheDocument();
+  });
+
+  // Use findByText to find the headers of each of the 4 columns
+  expect(await screen.findByText('Email Address')).toBeInTheDocument();
+  expect(await screen.findByText('In Use')).toBeInTheDocument();
+  expect(await screen.findByText('Overdue')).toBeInTheDocument();
+  expect(await screen.findByText('Role')).toBeInTheDocument();
+  
+  useAuthMock.mockRestore();
+});
+
+  it('fetches and displays user data', async () => {
+    await act(async () => {
+      render(
+        <BrowserRouter>
+          <Users />
+        </BrowserRouter>
+      );
+    });
+  
+    //wait for page to be done loading  
+    await waitFor(() => {
+      expect(screen.queryByTestId('ball-triangle-loading')).not.toBeInTheDocument();
+      
+    });
+
+    // Assert that the user data is displayed
+    mockDishesStatusData.forEach(async (data) => {
+      expect(await screen.findByText(data.emailAddress)).toBeInTheDocument();
+      expect(await screen.findByText(data.inUse)).toBeInTheDocument();
+      expect(await screen.findByText(data.overdue)).toBeInTheDocument();
+      expect(await screen.findByText(data.role)).toBeInTheDocument();
+    });
+
+    // Clean up mocks
+    useAuthMock.mockRestore();
+  });
+});
+
+describe('Search Functionality', () => {
+  it('allows users to be searched by email', async () => {
+
+      //Mock the responses of the getDishesStatusForEachUser so that our mockData is displayed in the table
+      const mockGetDishesStatusForEachUser = adminApi.getDishesStatusForEachUser as jest.Mock;
+      mockGetDishesStatusForEachUser.mockImplementation(async (sessionToken)=> {
+        return mockUsers.map(user => ({
+          userId: user.userId,
+          emailAddress: user.emailAddress,
+          inUse: mockInUseResponse.find((data) => data.email == user.emailAddress)?.count, 
+          overdue: mockOverdueResponse.find((data) => data.email == user.emailAddress)?.count, 
+        }));
+      });
+
+      await act(async () => {
+        render(
+         
+        <BrowserRouter>
+          <Users />
+        </BrowserRouter>
+     
+        );
+      });
+    
+      //wait for page to be done loading  
+      await waitFor(() => expect(screen.queryByTestId('ball-triangle-loading')).not.toBeInTheDocument());
+
+      // Change the search input
+      fireEvent.change(screen.getByPlaceholderText('search email...'), { target: { value: 'hello' } });
+    
+      // Wait for the search results to update
+      await waitFor(() => {
+        expect(screen.getByText('hello@ualberta.ca')).toBeInTheDocument();
+        // Make sure other users are not displayed
+        mockUsers.forEach(user => {
+          if (!user.emailAddress.includes('hello')) {
+            expect(screen.queryByText(user.emailAddress)).not.toBeInTheDocument();
+          }
+        });
+      });
```

Comment:
instead of using comments, declare variables to hold the things you want to test, it makes it a lot easier to read (and also change in the future)
```suggestion
      await waitFor(() => {
      const currentUserEmail = 'hello@ualberta.ca'
        expect(screen.getByText(currentUserEmail)).toBeInTheDocument();
        const otherUsers = users.filter(user => user.emailAddress !== currentUserEmail);
        otherUsers.forEach(user => {
            expect(screen.queryByText(user.emailAddress)).not.toBeInTheDocument();
      });
```